lib/generator.js

```diff
@@ -275,22 +274,65 @@ class Generator {
    * const Generator = require('asyncapi-generator');
    * const content = await Generator.getTemplateFile('html', '.partials/content.html');
    *
-   * @example <caption>Obtaining the content of a file from a custom template</caption>
-   * const path = require('path');
-   * const Generator = require('asyncapi-generator');
-   * const content = await Generator.getTemplateFile('myTemplate', 'a/file.js', {
-   *   templatesDir: path.resolve(__dirname, 'my-templates')
-   * });
-   *
    * @static
    * @param {String} templateName  Name of the template to generate.
    * @param {String} filePath      Path to the file to render. Relative to the template directory.
    * @param {Object} options
-   * @param {String} [options.templatesDir]  Path to the directory where to find the given template. Defaults to internal `templates` directory.
    * @return {Promise}
    */
-  static async getTemplateFile(templateName, filePath, { templatesDir } = {}) {
-    return await readFile(path.resolve(templatesDir || DEFAULT_TEMPLATES_DIR, templateName, filePath), 'utf8');
+  static async getTemplateFile(templateName, filePath) {
+    return await readFile(path.resolve(DEFAULT_TEMPLATES_DIR, templateName, filePath), 'utf8');
+  }
+
+  /**
+   * Downloads and installs a template and its dependencies.
+   *
+   * @param {Boolean} [force=false] Whether to force installation (and skip cache) or not.
+   */
+  installTemplate(force = false) {
+    return new Promise(async (resolve, reject) => {
+      if (!force) {
+        try {
+          let installedPkg;
+
+          if (isFileSystemPath(this.templateName)) {
+            const pkg = require(path.resolve(this.templateName, 'package.json'));
+            installedPkg = require(path.resolve(DEFAULT_TEMPLATES_DIR, pkg.name, 'package.json'));
+            if (installedPkg.name !== pkg.name) throw new Error('Package not installed');
```

Comment:
can you explain when it can happen as I could not figure out? and why we throw error instead of info + run installation?

Reply:
Good catch. This is not needed anymore. The reason it was throwing the error is for the parent try/catch to catch the error and avoid calling the `resolve` function a few lines below.